crates/pop-drink/src/error.rs

```diff
@@ -0,0 +1,258 @@
+//! A set of errors used for testing smart contracts.
+
+use std::fmt::Debug;
+
+pub use drink::{
+	pallet_assets::Error as AssetsError, pallet_balances::Error as BalancesError,
+	pallet_contracts::Error as ContractsError,
+};
+pub use pop_api::primitives::v0::*;
+use scale::{Decode, Encode};
+
+use crate::utils::decode;
+
+/// Configuration for the runtime error.
+pub trait RuntimeErrorConfig: Debug {
+	/// Error type of the runtime modules.
+	/// Reference: https://paritytech.github.io/polkadot-sdk/master/solochain_template_runtime/enum.RuntimeError.html.
+	type ModuleError: Decode + Encode + Debug;
+	/// Error type of the Pop API depends on the version.
+	type PopApiError: Decode + Encode + Debug + From<u32> + Into<u32>;
+	/// Index of the variant `RuntimeError::Module`. This is based on the index of
+	/// `PopApiError::Module`.
+	const MODULE_INDEX: u8;
+}
+
+/// Runtime error for testing.
+#[derive(Encode, Decode, Debug)]
+pub enum RuntimeError<T: RuntimeErrorConfig> {
```

Comment:
```suggestion
pub enum Error<T: RuntimeErrorConfig> {
```

Do we need to prefix it with Runtime? Is it not somewhat implied being that drink does exactly that? The variants arguably describe it sufficiently.

Reply:
Resolved